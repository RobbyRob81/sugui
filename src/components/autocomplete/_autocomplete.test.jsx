import * as React from 'react';
import { customRender, fireEvent } from '../../../test/test-utils';
import AutoComplete from '.';

const data = [{
  id: 'peter.pan@domain.com',
  label: 'Peter Pan'
}, {
  id: 'ai.weiwei@domain.com',
  label: 'Ai Weiwei'
}, {
  id: 'rafael.nadal@domain.com',
  label: 'Rafael Nadal'
}];

it('should render without throwing an error', () => {
  const { container } = customRender(<AutoComplete />);
  expect(container.firstChild).toMatchSnapshot();
});

it('Should allow the user to type a word to search for', async () => {
  const { getByPlaceholderText } = customRender(<AutoComplete placeholder="test" />);
  const inputNode = getByPlaceholderText('test');

  fireEvent.change(inputNode, { target: { value: 'lion' } });
  expect(inputNode.value).toBe('lion');
});

// it('calls "onSelect" prop when an item is selected', async () => {
//   const onSelect = jest.fn();
//   const { getByPlaceholderText, container } = customRender(<AutoComplete placeholder="test" data={data} onSelect={onSelect} />);
//   const inputNode = getByPlaceholderText('test');
//
//   // This will write `pete` in the input field
//   fireEvent.change(inputNode, { target: { value: 'pete' } });
//
//   const items = await waitForElement(() => {
//     container.querySelectorAll('.react-autosuggest__suggestion');
//   });
//
//   // TODO pending to fix timeout searching for the suggestion item...
//   // https://spectrum.chat/react-testing-library/help?thread=68e5c2dc-21fa-4b8d-9c6d-3bf2c0613c4a
//   // https://github.com/moroshko/react-autosuggest/issues/591
//   console.log(items);
//
//   // Fire the click on the first item
//   // fireEvent.click(items[0]);
//   expect(onSelect).toHaveBeenCalled();
// });
