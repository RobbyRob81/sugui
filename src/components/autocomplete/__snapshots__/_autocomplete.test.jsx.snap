// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div
  class="sc-gqjmRU gBaGjA"
>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-owns="react-autowhatever-1"
    class="react-autosuggest__container"
    role="combobox"
  >
    <div>
       
      <div
        class="sc-EHOje iTZenT"
      >
        <input
          aria-autocomplete="list"
          aria-controls="react-autowhatever-1"
          autocomplete="off"
          class="react-autosuggest__input sc-bZQynM gDpbKP"
          color="mineShaft"
          placeholder=""
          type="text"
          value=""
        />
        <div
          class="sc-htoDjs kCbvim"
          color="mineShaft"
        >
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h20v20H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M12.041 3.792a5.833 5.833 0 1 1-8.25 8.25 5.833 5.833 0 0 1 8.25-8.25M12.04 12.04l5.877 5.877"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </svg>
        </div>
      </div>
    </div>
    <span
      class="sc-VigVT iuLcPK"
      color="mineShaft"
    >
      Type 3 letters or more
    </span>
    <ul
      class="sc-jTzLTM lcGxWJ"
    >
      <li
        class="sc-fjdhpX bNcSeZ"
        color="mineShaft"
      />
      <li
        class="sc-fjdhpX bNcSeZ"
        color="mineShaft"
      />
      <li
        class="sc-fjdhpX bNcSeZ"
        color="mineShaft"
      />
    </ul>
    <div
      class="react-autosuggest__suggestions-container"
      id="react-autowhatever-1"
      role="listbox"
    >
      <section
        class="sc-jzJRlG lfEMQK sc-gZMcBi kvUJhL"
      />
    </div>
  </div>
</div>
`;
